### DISCUSSION TITLE
Fully support to NAOT

### DISCUSSION THREAD

**Discussion by leowest77** [stride3d/stride#2207] [General]
Do you guys have any plans to support NAOT or have a timeline or at least a list of things that needs to be done in order for stride to fully support NAOT?

Sorry about the direct question, its a very cool project seeing its using .net 8 I wonder how likely its to anytime soon support NAOT fully.

**Comment by IXLLEGACYIXL** [comment]
https://github.com/stride3d/stride/issues/2069

**Reply by leowest77** [reply]
ah thanks that's why I couldn't find it I was searching for NAOT.

**Comment by IXLLEGACYIXL** [comment]
the only benefit that aot would give is to compile it for playstation or consoles directly but aot doesnt support that yet afaik and xbox should be supported by microsoft anyway but isnt as UWP died so lets see what the future gives

**Reply by MeharDT** [reply]
I think FNA supports consoles via NativeAOT. For Xbox, Win32 and GDK seem to be the "successor" for UWP, I don't know if any Stride developers have experimented with it yet.

**Reply by leowest77** [reply]
godot also supports NAOT https://godotengine.org/article/platform-state-in-csharp-for-godot-4-2/#summary

**Reply by ericwj** [reply]
NAOT also protects your IP which is a huge reason to consider this vital.

It might also reduce deployment size by a lot - at least looking at code binaries alone, improve startup time and even improve steady state performance with some clever tricks like eager initialization of statics.

Provided that the latter benefit might require deploying at least two binaries, one with support for the latest intrinsics hardcoded and the other with runtime switching (based on `CPUID` indirectly) in order to support old processors.
